math-libs/BLAS/CMakeLists.txt

```
@@ -0,0 +1,296 @@
+set(_blas_subproject_names)
+
+therock_cmake_subproject_declare(hipBLAS-common
+EXTERNAL_SOURCE_DIR "hipBLAS-common"
+BACKGROUND_BUILD
+CMAKE_ARGS
+    -DROCM_PATH=
+    -DROCM_DIR=
+COMPILER_TOOLCHAIN
+    amd-hip
+IGNORE_PACKAGES
+    # The current version of rccl needs to download a 2y old version of rocm-cmake
+    # to work and it will only do so if the system resolver reports it not found
+    # without any other error (which due to the signature, our resolver will
+    # hard fail). Once fixed, `rocm-core` should be added to the BUILD_DEPS
+    # and this removed: https://github.com/nod-ai/TheRock/issues/18
+    ROCM
+RUNTIME_DEPS
+    hip-clr
+)
+therock_cmake_subproject_glob_c_sources(hipBLAS-common
+SUBDIRS
+    library
+)
+therock_cmake_subproject_provide_package(hipBLAS-common hipblas-common lib/cmake/hipblas-common)
+therock_cmake_subproject_activate(hipBLAS-common)
+list(APPEND _blas_subproject_names hipBLAS-common)
+
+##############################################################################
+# hipBLASLt
+##############################################################################
+
+therock_cmake_subproject_declare(hipBLASLt
+EXTERNAL_SOURCE_DIR "hipBLASLt"
+BACKGROUND_BUILD
+CMAKE_ARGS
+    -DHIP_PLATFORM=amd
+    -DROCM_PATH=
+    -DROCM_DIR=
+    -DTensile_LOGIC=
+    -DTensile_CODE_OBJECT_VERSION=default
+    -DTensile_CPU_THREADS=
+    -DTensile_LIBRARY_FORMAT=msgpack
+    -DTensile_HIP_CONFIG=${_blas_hack_hip_dir}/bin/hipconfig
+    "-DTensile_C_COMPILER=${_toolchain_dir}/bin/clang"
+    "-DTensile_CXX_COMPILER=${_toolchain_dir}/bin/clang++"
+    "-DTensile_ASSEMBLER=${_toolchain_dir}/bin/clang++"
+    "-DTensile_OFFLOAD_BUNDLER=${_toolchain_dir}/bin/clang-offload-bundler"
+    # TODO(#55): Enable once roctracer is added
+    -DHIPBLASLT_ENABLE_MARKER=OFF
+    # TODO: Enable clients.
+COMPILER_TOOLCHAIN
+    amd-hip
+BUILD_DEPS
+    hipBLAS-common
+    therock-msgpack-cxx
+RUNTIME_DEPS
+    hip-clr
+)
+therock_cmake_subproject_glob_c_sources(hipBLASLt
+SUBDIRS
+    .
+)
+therock_cmake_subproject_provide_package(hipBLASLt hipblaslt lib/cmake/hipblaslt)
+therock_cmake_subproject_activate(hipBLASLt)
+list(APPEND _blas_subproject_names hipBLASLt)
+
+
+##############################################################################
+# rocBLAS
+##############################################################################
+
+therock_cmake_subproject_declare(rocBLAS
+EXTERNAL_SOURCE_DIR "rocBLAS"
+BACKGROUND_BUILD
+CMAKE_ARGS
+    -DHIP_PLATFORM=amd
+    -DROCM_PATH=
+    -DROCM_DIR=
+    -DBUILD_WITH_TENSILE=ON
+    -DBUILD_WITH_HIPBLASLT=ON
+    # TODO: With `Tensile_TEST_LOCAL_PATH` set, the resulting build path is ${Tensile_TEST_LOCAL_PATH}/build.
+    "-DTensile_TEST_LOCAL_PATH=${CMAKE_CURRENT_SOURCE_DIR}/Tensile"
+    # # TODO: Enable clients.
+IGNORE_PACKAGES
+    ROCM
+COMPILER_TOOLCHAIN
+    amd-hip
+BUILD_DEPS
+    therock-msgpack-cxx
+    hipBLAS-common
+RUNTIME_DEPS
+    hip-clr
+    hipBLASLt
+)
```

nit: indentation is also inconsistent here and in a few other spots in the file. Two space indentation for all levels? This uses 0 indentation for each argument group and 4 spaces for the arguments
```suggestion
therock_cmake_subproject_declare(rocBLAS
  EXTERNAL_SOURCE_DIR "rocBLAS"
  BACKGROUND_BUILD
  CMAKE_ARGS
    -DHIP_PLATFORM=amd
    -DROCM_PATH=
    -DROCM_DIR=
    -DBUILD_WITH_TENSILE=ON
    -DBUILD_WITH_HIPBLASLT=ON
    # TODO: With `Tensile_TEST_LOCAL_PATH` set, the resulting build path is ${Tensile_TEST_LOCAL_PATH}/build.
    "-DTensile_TEST_LOCAL_PATH=${CMAKE_CURRENT_SOURCE_DIR}/Tensile"
    # # TODO: Enable clients.
  IGNORE_PACKAGES
    ROCM
  COMPILER_TOOLCHAIN
    amd-hip
  BUILD_DEPS
    therock-msgpack-cxx
    hipBLAS-common
  RUNTIME_DEPS
    hip-clr
    hipBLASLt
)
```

---

Will fix indentation throughout. For some reason, the cmake formatter badly messes up copy/paste indentation. Missed some.